release.js

```diff
@@ -0,0 +1,520 @@
+  if (ok) {
+    console.log('✅ Great!\n')
+  } else {
+    console.error('❌ Things were not correct. I don\'t know how to automate this case 🤖😞')
+    exitRelease()
+  }
+}
+
+const spawnAssumeOkay = async (cmd, cmdArgs, verbose) => {
+  const spinner = ora([cmd].concat(cmdArgs).join(' '))
+  if (!verbose) {
+    spinner.start()
+  }
+
+  let exitInProcess = false
+  const handleExit = error => {
+    if (exitInProcess) return
+    exitInProcess = true
+
+    spinner.fail()
+    somethingWentWrong(cmd)
+    if (error) {
+      console.error(`${error}\n`)
+    }
+    exitRelease()
+  }
+
+  await new Promise(resolve => {
+    const handle = spawn(cmd, cmdArgs, { cwd: '.' })
+    if (verbose) {
+      handle.stdout.pipe(process.stdout);
+    }
+    handle.stderr.pipe(process.stderr);
+
+    const onClose = code => {
+      if (code === 0) {
+        spinner.succeed()
+        resolve()
+      } else {
+        handleExit()
+      }
+    }
+    handle.on('close', onClose)
+    handle.on('exit', onClose)
+
+    handle.on('error', handleExit)
+  })
+}
+
+const execAssumeOkay = async cmd => {
+  const spinner = ora(cmd).start()
+
+  try {
+    const ret = await promiseExec(cmd)
+    spinner.stop()
+    return ret
+  } catch (error) {
+    spinner.stop()
+    somethingWentWrong(cmd)
+    console.error(error)
+    exitRelease()
+  }
+}
+
+const somethingWentWrong = (cmd) => {
+  console.error('❌ Oops. Something went wrong with that last command! 🤖😞 \n')
+  console.error(`❌ The command was: ${chalk.magenta(cmd)} \n`)
+}
+
+const checkWorkspaceIsClean = async () => {
+  const { stdout: workspaceIsUnclean } = await execAssumeOkay('git diff-index --quiet HEAD -- || echo "not clean"')
+  if (workspaceIsUnclean) {
+    console.error('❌ Your git workspace must be clean before starting a release 🤖😞')
+    exitRelease()
+  }
+  safeToClearWorkspace = true
+}
+
+const replaceInFile = (file, regex, replaceFunc) => {
+  fs.readFile(file, 'utf8', (err, data) => {
+    if (err) {
+      console.error('❌ Something went wrong trying to load the file!')
+      console.error(err)
+      exitRelease()
+    }
+
+    const result = data.replace(regex, replaceFunc)
+
+    fs.writeFile(file, result, 'utf8',  (err) => {
+       if (err) {
+         console.error('❌ Something went wrong trying to write to the file!')
+         console.error(err)
+         exitRelease()
+       }
+    })
+  })
+}
+
+const readInFile = async (file, regex, callback) => {
+  fs.readFile(file, 'utf8', (err, data) => {
+    if (err) {
+      console.error('❌ Something went wrong trying to load the file!')
+      console.error(err)
+      exitRelease()
+    }
+    const result = data.match(regex)
+    callback(result)
+  })
+}
+
+const execWithErrorHandling = async (cmd, callback) => {
+  const spinner = ora(cmd).start()
+
+  try {
+    const ret = await promiseExec(cmd)
+    spinner.stop()
+    return ret
+  } catch (error) {
+    spinner.stop()
+    callback()
```

Comment:
you could rethrow here, so that it's more inline with a promise behaviour. right not it's a bit of a mix of old and new ways